cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/BrokerSetAwareGoal.java

```diff
@@ -217,7 +225,9 @@ protected void rebalanceForBroker(Broker broker, ClusterModel clusterModel, Set<
           _replicaToBrokerSetMappingPolicy.brokerSetIdForReplica(replica, clusterModel, _brokerSetResolutionHelper);
 
       // If the re-balancing action is performed within the broker set boundary, we are good
-      if (broker.isAlive() && eligibleBrokerSetIdForReplica.equals(currentBrokerSetId)) {
+      // Also, if the topics that are configured to have leaders on each broker within the cluster, then those topics won't be rebalanced
+      if ((broker.isAlive() && eligibleBrokerSetIdForReplica.equals(currentBrokerSetId)) || _mustHaveTopicLeadersPerBroker.contains(
```

Comment:
@CCisGG  I don't think we need the logic after OR ``_mustHaveTopicLeadersPerBroker.contains(``

Based on your testing, I can remove this.

Reply:
I tested with that OR and it works good so far, but I think you're right that the OR logic is not needed here. Those must-have-leader topics should be in the excludedTopics which don't participate in the rebalance. I agree that we should remove it.

Reply:
Removed